StRoot/StFcsTrackMatchMaker/StFcsTrackMatchMaker.cxx

```diff
@@ -0,0 +1,234 @@
+    mNtrk[0]= new TH1F("NTrk_Ecal","NTrk_Ecal",10,0.0,10.0);
+    mNclu[0]= new TH1F("NEcalClu_Trk","NEcalClu_Trk",10,0.0,10.0);
+
+    mHdx[1] = new TH1F("dx_HcalTrk","dx Hcal-Track",100,-50,50);
+    mHdy[1] = new TH1F("dy_HcalTrk","dy Hcal-Track",100,-50,50);	
+    mHdr[1] = new TH1F("dr_HcalTrk","dr Ecal-Track",100,0,20);
+    mNtrk[1]= new TH1F("NTrk_Hcal","NTrk_Hcal",10,0.0,10.0);
+    mNclu[1]= new TH1F("NHcalClu_Trk","NHcalClu_Trk",10,0.0,10.0);
+
+    mNtrk[2]= new TH1F("NTrk","NTrk/evt",50,0.0,100.0);
+    mNtrk[3]= new TH1F("NGoodTrk","NGoodTrk/evt",20,0.0,20.0);
+
+    mNclu[2]= new TH1F("NEcalClu","NEcalClu/evt",30,0.0,30.0);
+    mNclu[3]= new TH1F("NHcalClu","NHcalClu/evt",30,0.0,30.0);
+
+    mPtEt[0] =new TH1F("ETovPT_E","ETovPT_E",50,0.0,2.0);
+    mPtEt[1] =new TH1F("ETovPT_EH","ETovPT_E+H",50,0.0,2.0);
+    mPtEt2[0]=new TH2F("ETPT_E", "ETvsPT_E; ET(Ecal); TrkPT",50,0.0,8.0,50,0.0,8.0);
+    mPtEt2[1]=new TH2F("ETPT_EH","ETvsPT_E+H ET(E+H); TrkPT",50,0.0,8.0,50,0.0,8.0);
+
+    mCharge[0]=new TH1F("Charge_E","Charge_E",3,-1.5,1.5);
+    mCharge[1]=new TH1F("Charge_H","Charge_H",3,-1.5,1.5);
+    mCharge[2]=new TH1F("Charge","Charge",3,-1.5,1.5);
+
+    mXY[0]=new TH2F("XY_E","XY_E",50,-130,130,50,-110,110);
+    mXY[1]=new TH2F("XY_H","XY_H",50,-130,130,50,-110,110);
+    mXY[2]=new TH2F("XY","XY",50,-130,130,50,-110,110);
+  }
+  return kStOK;
+}
+
+int StFcsTrackMatchMaker::Finish(){
+  if(mEpdgeo) delete mEpdgeo;
+  if(mFile){
+    LOG_INFO << "Closing "<<mFilename<<endm;
+    mFile->Write();
+    mFile->Close();
+  }
+  return kStOK;
+}
+
+int StFcsTrackMatchMaker::Make(){
+  StEvent* event = (StEvent*)GetInputDS("StEvent");
+  if(!event) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent"<<endm; 
+    return kStErr;
+  }
+  mFcsColl = event->fcsCollection();
+  if(!mFcsColl) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent->StFcsCollection"<<endm; 
+    return kStErr;
+  }
+  mFwdTrkColl = event->fwdTrackCollection();
+  if(!mFwdTrkColl) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent->fwdTrackCollection"<<endm; 
+    return kStErr;
+  }
+    
+  int ntrk=mFwdTrkColl->numberOfTracks();
+  int ngoodtrk=0;
+  int nMatch[2]={0,0};
+  for(int itrk=0; itrk<ntrk; itrk++){
+    StFwdTrack* trk=mFwdTrkColl->tracks()[itrk];
+    if(trk->didFitConvergeFully()==false) continue;
+    ngoodtrk++;
+    if(Debug()){
+      LOG_INFO << Form("Proj0 : %6.2f %6.2f %6.2f",trk->mProjections[0].XYZ.x(),trk->mProjections[0].XYZ.y(),trk->mProjections[0].XYZ.z())<<endm;
+      LOG_INFO << Form("Proj1 : %6.2f %6.2f %6.2f",trk->mProjections[1].XYZ.x(),trk->mProjections[1].XYZ.y(),trk->mProjections[1].XYZ.z())<<endm;
+      LOG_INFO << Form("Proj2 : %6.2f %6.2f %6.2f",trk->mProjections[2].XYZ.x(),trk->mProjections[2].XYZ.y(),trk->mProjections[2].XYZ.z())<<endm;
+    }
+
+    //North or south from track
+    int ns=0;
+    if(trk->mProjections[0].XYZ.x()>0.0) ns=1;
```

Comment:
mProjections should not be public on StFwdTrack.

Reply:
What is wrong in this case by making it public on StFwdTrack?

Reply:
I believe our coding standards require member variables to not be publicly accessible.

Reply:
The coding guidelines suggest to make data members private except in structs https://star-bnl.github.io/star-coding-guidelines/codingguide.xml#Access_Control
It also looks like Akio has fixed this in the followup commit. However, I don't see where the `StFwdTrack::mProjections` are being set in the submitted code. There appears to be no setter.